File: app/src/main/java/com/lyl/game/GameBodyFactory.java
package com.lyl.game;

import com.lyl.game.impl.SnakeImpl;
import com.lyl.game.interfaces.IGameBody;
import com.lyl.game.interfaces.ISnakeCallback;

/**
 * create lyl on 2019-08-16
 * </p>
 */
public class GameBodyFactory {

    public static final int GAME_SNAKE = 1;

    public static IGameBody createGameBody(int type, ISnakeCallback callback) {
        IGameBody gameBody = null;
        switch (type) {
            case GAME_SNAKE:
                gameBody = new SnakeImpl(callback);
                break;
            case 2:
                break;
            default:
                break;
        }
        return gameBody;
    }
}

File: app/src/main/java/com/lyl/game/GameBodyFactory.kt
package com.lyl.game

import com.lyl.game.impl.SnakeImpl
import com.lyl.game.interfaces.IGameBody
import com.lyl.game.interfaces.ISnakeCallback

/**
 * create lyl on 2019-08-16
 *
 */
object GameBodyFactory {

    const val GAME_SNAKE = 1

    fun createGameBody(type: Int, callback: ISnakeCallback): IGameBody? {
        var gameBody: IGameBody? = null
        when (type) {
            GAME_SNAKE -> gameBody = SnakeImpl(callback)
            else -> {
            }
        }
        return gameBody
    }
}
